src/ObjectUtils/CoffFile.cpp

```diff
@@ -366,6 +369,94 @@ bool CoffFileImpl::HasDebugSymbols() const {
+    return ErrorMessage{
+        absl::StrFormat("Unable to assign sizes to symbols from the Export Table: %s",
+                        unwind_ranges_or_error.error().message())};
+  }
+  absl::flat_hash_map<uint64_t, uint64_t> unwind_range_start_to_size;
+  for (const UnwindRange& unwind_range : unwind_ranges_or_error.value()) {
+    unwind_range_start_to_size.emplace(unwind_range.start, unwind_range.end - unwind_range.start);
+  }
+
+  ModuleSymbols module_symbols;
+  for (const llvm::object::ExportDirectoryEntryRef& ref : object_file_->export_directories()) {
+    // From https://docs.microsoft.com/en-us/windows/win32/debug/pe-format#export-address-table:
+    // "A forwarder RVA exports a definition from some other image, making it appear as if it were
+    // being exported by the current image." So these symbols don't really belong to this file, and
+    // we skip them.
+    bool is_forwarder{};
+    if (llvm::Error error = ref.isForwarder(is_forwarder)) {
+      llvm::consumeError(std::move(error));
+      continue;
+    }
+    if (is_forwarder) continue;
+
+    uint32_t rva{};
+    if (llvm::Error error = ref.getExportRVA(rva)) {
+      llvm::consumeError(std::move(error));
+      continue;
+    }
+    const uint64_t virtual_address = object_file_->getImageBase() + rva;
+
+    std::string name;
+    llvm::StringRef name_ref;
+    // From the documentation of ExportDirectoryEntryRef::getSymbolName: "If the symbol is exported
+    // only by ordinal, the empty string is set as a result." However, if the name pointer table is
+    // empty, the error "Invalid data was encountered while parsing the file" is produced instead.
+    // So consider both cases.
+    if (llvm::Error get_symbol_name_error = ref.getSymbolName(name_ref);
+        !get_symbol_name_error && !name_ref.empty()) {
+      name = name_ref.str();
+    } else {
+      if (get_symbol_name_error) llvm::consumeError(std::move(get_symbol_name_error));
+      uint32_t ordinal{};
+      if (llvm::Error get_ordinal_error = ref.getOrdinal(ordinal)) {
+        llvm::consumeError(std::move(get_ordinal_error));
+        continue;
+      }
+      // We arbitrarily choose NONAME# because functions exported only by ordinal are specified in
+      // the .def file with the NONAME attribute. See
+      // https://docs.microsoft.com/en-us/cpp/build/exporting-functions-from-a-dll-by-ordinal-rather-than-by-name
+      name = "NONAME" + std::to_string(ordinal);
+    }
+
+    SymbolInfo symbol_info;
+    symbol_info.set_demangled_name(std::move(name));
+    symbol_info.set_address(virtual_address);
+
+    // The Export Table doesn't contain the size of symbols: obtain this size from the Exception
+    // Table, but note that that's not always possible, as leaf functions have no RUNTIME_FUNCTION.
+    auto unwind_range_start_to_size_it = unwind_range_start_to_size.find(virtual_address);
+    if (unwind_range_start_to_size_it == unwind_range_start_to_size.end()) {
+      symbol_info.set_size(0);
```

Comment:
I am wondering if we should actually do that. If we don't have a size, we won't be able to do much with the symbol. So we might want to discard it. WDYT?

Reply:
First of all, this is consistent with `DeduceDebugSymbolMissingSizesAsDistanceFromNextSymbol`, where we set the size to 0 if it's still unknown: this way at least the symbol shows up at all in the function list, even if its size is wrong.

However, the practical reason is that the idea is that these will be merged with the symbols from unwind info. Once that's the case, we can set these zeros back to `kUnknownSymbolSize`, call `DeduceDebugSymbolMissingSizesAsDistanceFromNextSymbol`, and get mostly correct sizes (unless there's a big gap between an exported symbol and the following symbol from the unwind info, or there's a leaf function without unwind info after the exported symbol).

Reply:
Sounds good. But I would at least add a comment regarding your second part.

Reply:
The way this function is laid out right now, it's also possible to use it "standalone". The explanation I gave you will be in the part that puts this and the unwind info together, i.e., a few pull requests from now.